File: #2044.count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) {

        map<int, int> m;
        int mx = 0;
        for (int num: nums) {
            map<int, int> tems(m);
            for(auto tem: tems){
                m[num | tem.first] += tem.second;
            }
            m[num] += 1;
            mx = mx | num;
        }
        return m[mx];
    }
};

File: #2044.count-number-of-maximum-bitwise-or-subsets.ts
function countMaxOrSubsets(nums: number[]): number {
    function countSubSets(nums, index, curOr, tarOr) {
        if (index == nums.length) {
            return curOr == tarOr ? 1 : 0;
        }

        const withoutCur = countSubSets(nums,index+1,curOr,tarOr);
        const withCur = countSubSets(nums,index+1, curOr|nums[index],tarOr);
        return withCur + withoutCur;
    }

    let maxOr = 0;
    for (let num of nums) {
        maxOr |= num;
    }
    return countSubSets(nums, 0, 0, maxOr);
};
